### PR TITLE
Refactors calls to languages.setLanguageConfiguration to declarative descriptions in language-configuration.json

### LINKED ISSUE #98621: Activate embedded languages for indent rules
From https://github.com/microsoft/vscode/pull/98181#issuecomment-633875186

> PHP defines HTML as an embedded language here: https://github.com/microsoft/vscode/blob/master/extensions/php/package.json#L44. The bug is that VSCode doesn't activate 'html' when it needs it's indent rules. As HTML defines the indent rules in code, activation is necessary.

> It's a VSCode issue. I suggest to create a new issue and assign it to @alexdima.

### PR DESCRIPTION
**Pull request by hediet**
This fixes #98621.

**Edit from @alexdima:** Since a couple of milestones, it is possible to express everything in the language configuration files, including the `wordPattern` and the `onEnterRules`. It is preferable to express the language configuration in JSON because this allows those features to work even when the main extension is not activated.

### CODE REVIEW

**Review by alexdima — APPROVED** [review]

File: extensions/handlebars/language-configuration.json [hunk 1]
```diff
@@ -22,5 +22,24 @@
 		{ "open": "\"", "close": "\"" },
 		{ "open": "<", "close": ">" },
 		{ "open": "{", "close": "}" }
-	]
+	],
+	"wordPattern": {
+		"pattern": "(-?\\d*\\.\\d\\w*)|([^\\`\\~\\!\\@\\$\\^\\&\\*\\(\\)\\=\\+\\[\\{\\]\\}\\\\\\|\\;\\:\\'\\\"\\,\\.\\<\\>\\/\\s]+)",
+		"flags": "g"
```
Comment (aeschli) [inline extensions/handlebars/language-configuration.json:28]: I believe the 'g' is mandatory for all word patterns. I has to do with the way we use the word pattern regexes.
So I would remove it from the language-configurations and set it where we create the RegExp object.

So the extension can do `"wordPattern": "(-?\\d..." *` (unless `i` or `u` is needed as well)

**Review by aeschli — APPROVED** [review]
As mentioned, I would remove the `g` from the wordPatterns in the language-configuration. Other than that. it all looks good!

**Review by roblourens — APPROVED** [review]

**Review by mjbvz — APPROVED** [review]
Markdown changes look good. Should TypeScript also be updated?

https://github.com/microsoft/vscode/blob/d89e2e17a5d1ba37c99b1d3929eb6180a5bfc7a8/extensions/typescript-language-features/src/languageFeatures/languageConfiguration.ts#L58

**Comment by hediet** [comment]
> Markdown changes look good. Should TypeScript also be updated?
> 
> https://github.com/microsoft/vscode/blob/d89e2e17a5d1ba37c99b1d3929eb6180a5bfc7a8/extensions/typescript-language-features/src/languageFeatures/languageConfiguration.ts#L58

I guess yes, but lets do that in another PR, otherwise this becomes a Frankenstein!
The typescript-basics language feature currently does not have separate language-configurations for ts and for tsx, however, the typescript-language-features registers different configurations for them.

**Review by roblourens — APPROVED** [review]

**Pull request merged**